### PR TITLE
NWB export, shorten subj key, PEP8 on schemas

### PR DESCRIPTION
**Pull request by CBroz1**

### CODE REVIEW

**Review by dimitri-yatsenko — APPROVED** [review]

File: element_animal/export/nwb.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        subject_id=subject_info.pop('subject'),
+        sex=subject_info.pop('sex'),
+        date_of_birth=datetime.combine(subject_info.pop('subject_birth_date'),
+                                       datetime.strptime('00:00:00', '%H:%M:%S'
+                                                         ).time()),
+        description=json.dumps(subject_info, default=str),
+        genotype=' x '.join((subject.Line.Allele * subject.Subject.Line
+                             & subject_key
+                             ).fetch('allele')),
+        species=subject_info.pop('species')
```
Comment (dimitri-yatsenko) [inline element_animal/export/nwb.py:25]:
```suggestion
        subject_id=subject_info['subject'],
        sex=subject_info['sex'],
        date_of_birth=datetime.combine(subject_info['subject_birth_date'],
                                       datetime.strptime('00:00:00', '%H:%M:%S'
                                                         ).time()),
        description=json.dumps(subject_info, default=str),
        genotype=' x '.join((subject.Line.Allele * subject.Subject.Line
                             & subject_key
                             ).fetch('allele')),
        species=subject_info.pop['species']
```

**Comment by CBroz1** [comment]
Hi @kabilar - We discussed stashing nwb adds pending a U24-wide push. While I agree that that's the case with workflows that specifically highlight in-development functionality (e.g., workflow-session), I think this is ready to merge. It will do 2 things preemptively: 1) include pynwb in requirements, 2) provide a function that would go unused. But happy to refactor if you're looking to avoid those.

File: setup.py [hunk 2]
```diff
@@ -19,7 +19,7 @@
     version=__version__,
     description="DataJoint Element for Animal Management",
     long_description=long_description,
-    long_description_content_type = 'text/markdown',
+    long_description_content_type='text/markdown',
     author='DataJoint NEURO',
     author_email='info@vathes.com',
```
Comment (kabilar) [inline setup.py:24]:
```suggestion
    author='DataJoint',
    author_email='info@datajoint.com',
```

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+blank_issues_enabled: false
+contact_links:
+  - name: DataJoint Contribution Guideline
+    url: https://docs.datajoint.io/python/community/02-Contribute.html
```
Comment (kabilar) [inline .github/ISSUE_TEMPLATE/config.yml:4]:
```suggestion
    url: https://docs.datajoint.org/python/community/02-Contribute.html
```

**Comment by kabilar** [comment]
Hi @CBroz1, to clarify, are the Issue templates directly from `datajoint-python`?  Thanks.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Bug Report
+
+### Description
+A clear and concise description of what is the overall operation that is intended to be performed that resulted in an error.
+
+### Reproducibility
+Include:
+- OS (WIN | MACOS | Linux)
+- Pharus Version
```
Comment (kabilar) [inline .github/ISSUE_TEMPLATE/bug_report.md:18]:
```suggestion
- DataJoint Element-Animal Version
```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+### Alternative Considerations
+Do you currently have a work-around for this? Provide any alternative solutions or features you've considered.
+
+### Related Errors
+Add any errors as a direct result of not exposing this feature.
+
+Please include steps to reproduce provided errors as follows:
+- OS (WIN | MACOS | Linux)
+- Pharus Version
```
Comment (kabilar) [inline .github/ISSUE_TEMPLATE/feature_request.md:29]:
```suggestion
- DataJoint Element-Animal Version
```

File: element_animal/version.py [hunk 6]
```diff
@@ -1,2 +1,2 @@
 """Package metadata."""
-__version__ = '0.1.0b0'
\ No newline at end of file
+__version__ = '0.1.0b1'
```
Comment (kabilar) [inline element_animal/version.py:2]: Updating the version will trigger a tag and release.  Should we hold off on releasing until we have finalized the `element-array-ephys` NWB export function?

File: CHANGELOG.md [hunk 7]
```diff
@@ -2,11 +2,15 @@
 
 Observes [Semantic Versioning](https://semver.org/spec/v2.0.0.html) standard and [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) convention.
 
+## [0.1.0b1] - 2022-01-13
```
Comment (kabilar) [inline CHANGELOG.md:5]:
```suggestion
## [unreleased]
```

Change text above if we do not update the version in this pull request.

File: element_animal/subject.py [hunk 8]
```diff
@@ -126,9 +138,14 @@ class Lab(dj.Part):
...
         ---
-        subject_alias=''    : varchar(32)  # alias of the subject in this lab, if different from the id
+        subject_alias='' : varchar(32) # alias for lab, if different from id
         """
 
+    @classmethod
+    def make_nwb(cls, subject_key):
+        from .export import subject_to_nwb
+        return subject_to_nwb(subject_key)
+
```
Comment (kabilar) [inline element_animal/subject.py:148]: Hi @CBroz1 @ttngu207, why was this added as a method here?  Thanks.

File: requirements.txt [hunk 9]
```diff
@@ -1 +1,2 @@
 datajoint>=0.13
+pynwb
```
Comment (kabilar) [inline requirements.txt:2]: Should we version lock so that we know which version our export function is tested to work with?  If we are not ready to lock the version, I suggest creating a GitHub Issue to remind us.

File: element_animal/export/nwb.py [hunk 10]
```diff
@@ -0,0 +1,26 @@
...
+        sex=subject_info['sex'],
+        date_of_birth=datetime.combine(subject_info['subject_birth_date'],
+                                       datetime.strptime('00:00:00', '%H:%M:%S'
+                                                         ).time()),
+        description=json.dumps(subject_info, default=str),
+        genotype=' x '.join((subject.Line.Allele * subject.Subject.Line
+                             & subject_key
+                             ).fetch('allele')),
+        species=subject_info['species']
+    )
```
Comment (kabilar) [inline element_animal/export/nwb.py:26]: This looks good to me.  @ttngu207, you had this in a draft pull request (#12).  Any concerns with going forward with the merge?  Thanks.

File: requirements.txt (hunk 9 above)
Comment (ttngu207) [reply requirements.txt:2]: Yes I think we should version lock this, to whatever is the latest version now, after tests passed.

**Review by ttngu207 — APPROVED** [review]

File: element_animal/subject.py (hunk 8 above)
Comment (ttngu207) [reply element_animal/subject.py:148]: @kabilar, I don't fully understand the question here, how are we intending for users to use this `subject_to_nwb` ?

Comment (ttngu207) [reply element_animal/subject.py:148]: Do you also mean why does it have to be a `classmethod`?

Comment (kabilar) [reply element_animal/subject.py:148]: Thanks @ttngu207.  Yes, exactly, how would users use `make_nwb` and `subject_to_nwb`?  I presume we would use the export function by running `subject_to_nwb({'subject':<varchar(8)>})`.  So I am not sure what `make_nwb` does?

File: requirements.txt (hunk 9 above)
Comment (CBroz1) [reply requirements.txt:2]: My local install of pynwb 1.4.0 has passed all tests. The latest version is 2.0.0., which I recall having issues with dependency matching. During alpha testing, I can revisit this issue and document the nature of the conflict. For now, I would pin to 1.4.0

Comment (kabilar) [reply requirements.txt]: Thanks.  I have created Issue #18.

File: element_animal/subject.py (hunk 8 above)
Comment (kabilar) [reply element_animal/subject.py:148]: After discussion with @ttngu207, this method was intended for the table to be able to export its data with the `Subject.make_nwb()` command instead of `element_animal.export.nwb import subject_to_nwb`.  Since the `element-array-ephys` NWB export function does not seem to be implementing this strategy, let's remove for now to be consistent.

Comment (CBroz1) [reply element_animal/subject.py]: Gotcha, all set. Thanks for thinking this through
